### DISCUSSION TITLE
How to get PPF for minima?

### DISCUSSION THREAD

**Discussion by nucflash** [georgebv/pyextremes#90] [Q&A] [answered]
I'm plotting my data, and I want to add a horizontal line, say, at 75% of the maxima, and at 75% of the minima, using the respective fitted models for maxima and minima. 

I do
```
def eva(data, extreme_types):
    model = EVA(data=data)

    model.get_extremes(
        method="BM",
        extremes_type=extremes_type,
        block_size="365.2425D"
        errors="raise")
        
    model.fit_model()
    return model
    
model_high = eva(data, 'high')
model_low = eva(data, 'low')

high_75pct = model_high.distribution.distribution.ppf(0.75)
low_75pct = model_high.distribution.distribution.ppf(0.75)

assert high_75pct != low_75pct   # should not fail, but fails
```

`high_75pct`, and `low_75pct` come out exactly the same. I see that you do something with ranks in `get_return_periods` (L121-127), but I don't know how to tap into it. Any guidance is appreciated.

**Comment by georgebv** [comment]
This example is not reproducible because `data` is unknown.

You are also using `distribution` directly. Distribution is fit to transformed extremes - maxima left intact, minima flipped to behave like maxima. When `EVA` calculates return values it performs inverse transformation: https://github.com/georgebv/pyextremes/blob/3706ff8303027499001f0bdff5dcbe6082399a58/src/pyextremes/eva.py#L1253-L1259

**Reply by nucflash** [reply]
Thanks George! 

Happy to use some sample data to make this reproducible. Do you have a dataset to suggest? 

Do I interpret you correctly if I say that for maxima, `model.cdf()` should match the output from `model.distribution.distribution.cdfd()`?

**Reply by georgebv** [reply]
To add some context - transformation is necessary because GEV has specific shape which represents extreme maxima. For minima we need this inverted. There are two ways to achieve that - either implement different distributions for maximima/minima or simply transform minima to maxima (flip around mode) and use the same distribution (we need to transform results back when interpreting results because of that).

![scipy-stats-genextreme-1](https://github.com/user-attachments/assets/9f6ecc84-3f7c-461d-9718-46201b74f89c)

**Reply by georgebv** [reply]
You can use numpy to generate a preudo-random data (don't forget to set a seed value to make it reproducible randomness). You can use scipy to sample from a specific distribution.

**Reply by georgebv** [reply, marked as answer]
`eva.model.distribution` points to instance of class `Distribution` specific to `pyextremes`. `eva.model.distribution.distribution` points to `scipy.rv_continuous` distribution which has default values for its fit parameters (shapes, loc, scale). These are different classes from different libraries with different interfaces.

`eva.model.cdf(x)` and `eva.model.distribution.get_prop('cdf', x, eva.model.fit_parameters)` would give the same result.

Because of that, in your example you are calling the same default scipy distribution with default parameters so results are always going to be the same. `data` is irrelevant here.

**Reply by nucflash** [reply]
I see. That makes a lot of sense. So I always need to use `model.cdf()` or `model.distribution.get_prop()`. Thanks for the clarification! And thank you again for taking the time and patience to answer my questions!
